pkg/webhooks/validation/controller/controller.go

```diff
@@ -271,7 +271,9 @@ func (c *Controller) updateValidatingWebhookConfiguration(current *kubeApiAdmiss
 	}
 	updated := current.DeepCopy()
 	for i := range updated.Webhooks {
-		updated.Webhooks[i].ClientConfig.CABundle = caBundle
+		if caChangeNeeded {
```

Comment:
Is this a performance optimization or a bug fix? Will there be any side effects if there is no `if`?

Reply:
TBH I don't get how it can be either. Some more explanation here would help.

Reply:
It is neither a perf improvement nor a bug fix. It is just to follow the same pattern that we follow for failurePolicy

Reply:
the failure policy one is actually a correctness issue fwiw . I don't really mind it but it seems a bit more fragile given it's zero-risk and zero cost to always write the cabundle?

Reply:
> I don't really mind it but it seems a bit more fragile given it's zero-risk and zero cost to always write the cabundle?

Sure. Do not feel strongly about. Will close this PR